azure_storage/azure_storage_api.py

```diff
@@ -234,3 +235,44 @@ async def get_directories(container_client):
     except FolderListError as error:
         print(error)
         return []
+
+async def get_pipeline_info(
+        connection_string: str,
+        pipeline_container_name: str,
+        pipeline_version: str
+    ) -> json:
+    """
+    Retrieves the pipeline information from Azure Blob Storage based on the
+    provided parameters.
+
+    Args:
+        connection_string (str): The connection string for the Azure Blob
+        Storage. pipeline_container_name (str): The name of the container where
+        the pipeline files are stored. pipeline_version (str): The version of
+        the pipeline to retrieve.
+
+    Returns:
+        json: The pipeline information in JSON format.
+
+    Raises:
+        PipelineNotFoundError: If the specified version of the pipeline is not
+        found.
+    """
+    try:
+        blob_service_client = BlobServiceClient.from_connection_string(
```

Comment:
https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-client-management?tabs=python

> Manage client objects
> A best practice for Azure SDK client management is to treat a client as a singleton, meaning that a class will only have one object at a time. There's no need to keep more than one instance of a client for a given set of constructor parameters or client options. This concept can be implemented in many ways, including:

> Creating a single client object and passing it as a parameter throughout the application. This approach is shown in the code examples in this article.
> Storing a client instance in a field. To learn more about C# fields, see [Fields (C# Programming Guide)](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/fields).
> Registering the client object as a singleton in a dependency injection container of your choice. For more information on dependency injection in ASP.NET Core apps, see [Dependency injection with the Azure SDK for .NET]> (https://learn.microsoft.com/en-us/dotnet/azure/sdk/dependency-injection).
This approach is far more efficient at scale than calling a constructor for each client that you need.

Reply:
Done